### ISSUE TITLE
Crazy Logging I want to shut it down

### ISSUE THREAD

**Issue by lingjivoo** [NVIDIA-NeMo/NeMo#15615]
When I am doing transcribing, the annoying transcribing logging overwhelms my logging and make me crazy. I tried a lot of solution to shut it down, but I failed.

6-04-16 13:14:00 label_looping_base:109 rank:0] No conditional node support for Cuda.                                                   │··································································································································
    Cuda graphs with while loops are disabled, decoding speed will be slower                                                                       │··································································································································
    Reason: Driver supports cuda toolkit version 12.2, but the driver needs to support at least 12,3. Please update your cuda driver.              │··································································································································
[NeMo I 2026-04-16 13:14:02 rnnt_models:292 rank:0] Timestamps requested, setting decoding timestamps to True. Capture them in Hypothesis object,  │··································································································································
                       with output[0][idx].timestep['word'/'segment'/'char']                                                                       │··································································································································
[NeMo I 2026-04-16 13:14:02 rnnt_models:226 rank:0] Using RNNT Loss : tdt                                                                          │··································································································································
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}                            │··································································································································
[NeMo W 2026-04-16 13:14:02 label_looping_base:109 rank:0] No conditional node support for Cuda.                                                   │··································································································································
    Cuda graphs with while loops are disabled, decoding speed will be slower                                                                       │··································································································································
    Reason: Driver supports cuda toolkit version 12.2, but the driver needs to support at least 12,3. Please update your cuda driver.              │··································································································································
[NeMo I 2026-04-16 13:14:04 rnnt_models:292 rank:0] Timestamps requested, setting decoding timestamps to True. Capture them in Hypothesis object,  │··································································································································
                       with output[0][idx].timestep['word'/'segment'/'char']                                                                       │··································································································································
[NeMo I 2026-04-16 13:14:04 rnnt_models:226 rank:0] Using RNNT Loss : tdt                                                                          │··································································································································
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}                            │··································································································································
[NeMo W 2026-04-16 13:14:04 label_looping_base:109 rank:0] No conditional node support for Cuda.                                                   │··································································································································
    Cuda graphs with while loops are disabled, decoding speed will be slower                                                                       │··································································································································
    Reason: Driver supports cuda toolkit version 12.2, but the driver needs to support at least 12,3. Please update your cuda driver.              │··································································································································
[NeMo I 2026-04-16 13:14:05 rnnt_models:292 rank:0] Timestamps requested, setting decoding timestamps to True. Capture them in Hypothesis object,  │··································································································································
                       with output[0][idx].timestep['word'/'segment'/'char']                                                                       │··································································································································
[NeMo I 2026-04-16 13:14:05 rnnt_models:226 rank:0] Using RNNT Loss : tdt                                                                          │··································································································································
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}                            │··································································································································
[NeMo W 2026-04-16 13:14:05 label_looping_base:109 rank:0] No conditional node support for Cuda.                                                   │··································································································································
    Cuda graphs with while loops are disabled, decoding speed will be slower                                                                       │··································································································································
    Reason: Driver supports cuda toolkit version 12.2, but the driver needs to support at least 12,3. Please update your cuda driver.              │··································································································································
[NeMo I 2026-04-16 13:14:07 rnnt_models:292 rank:0] Timestamps requested, setting decoding timestamps to True. Capture them in Hypothesis object,  │··································································································································
                       with output[0][idx].timestep['word'/'segment'/'char']                                                                       │··································································································································
[NeMo I 2026-04-16 13:14:07 rnnt_models:226 rank:0] Using RNNT Loss : tdt                                                                          │··································································································································
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}                            │··································································································································
[NeMo W 2026-04-16 13:14:07 label_looping_base:109 rank:0] No conditional node support for Cuda.                                                   │··································································································································
    Cuda graphs with while loops are disabled, decoding speed will be slower                                                                       │··································································································································
    Reason: Driver supports cuda toolkit version 12.2, but the driver needs to support at least 12,3. Please update your cuda driver.              │··································································································································
                                                              

It is a good model, but this problem makes me annoyed.

**Comment by sakgoyal** [comment]
well if you're on an nvidia card, have you at least updated the cuda driver? that would fix at least some of those warnings

**Comment by azziko** [comment]
Try one of these https://github.com/NVIDIA-NeMo/NeMo/issues/1376

I used https://github.com/NVIDIA-NeMo/NeMo/issues/1376#issuecomment-3234432315

**Comment by pzelasko** [comment]
@artbataev can you take a look at some of these label looping / decoding logs? Maybe we should change them to logging level DEBUG?

**Comment by artbataev** [comment]
@lingjivoo What is your NeMo version? How are you using `transcribe`?

If I correctly understand from the logs, you are calling `model.transcribe(..., timestamps=True)` multiple times using some old NeMo version.
Your see these logs since each time you are calling `.transcribe`, the decoder is reinstantiated. This issue is fixed in https://github.com/NVIDIA-NeMo/NeMo/pull/15298 and should be fixed since NeMo 2.7.0.
You will see these logs only once, and I think that these logs are useful once you did not disable CUDA graphs explicitly.
If you want to disable them, use `. change_decoding_strategy `, see https://github.com/NVIDIA-NeMo/NeMo/issues/15145#issuecomment-4112885836

@lingjivoo Please, confirm that the issue is fixed in NeMo 2.7.0+

**Comment by rehan243** [comment]
So you're getting slammed with NeMo logging during transcription and can't seem to shut it down? What's your NeMo version, btw? We've had similar issues with logging verbosity in our production transcription pipeline using NeMo v1.23 - we ended up having to manually configure the logging levels for specific modules like `rnnt_models` and `label_looping_base`. Turns out, you can actually control the logging level by setting `logging.getLogger('nemo').setLevel(logging.WARNING)` or something similar. The thing is, NeMo's logging can be pretty finicky, and we've seen cases where certain warnings, like that CUDA driver version mismatch, are actually important. We've been using a custom logging config to filter out the noise while keeping the important stuff. Not gonna lie, it took us a few tries to get it right. We're on CUDA 12.4 now, and that particular warning went away after we updated our drivers.

**Comment by pzelasko** [comment]
Thanks for the feedback. It's likely we'll revise logging overall. What improvements would you suggest?

**Comment by rehan243** [comment]
hm, fair point on revising logging globally. yeah, for now, I’d probably lean on setting specific log levels per module, especially if you want to keep some debug info around without drowning in noise — like `nemo` or `rnnt_models`. oh wait, you're on v2.x or v3? some of the logging configs changed between versions, so might be worth double-checking the docs. lmk if you want a quick snippet for that!

**Comment by ethical-haquer** [comment]
Personally, this is the first Python library I've used where, out of the box, it logs what I would deem "DEBUG" level stuff to stdout.

I was able to prevent nearly 50 lines of logging going to stdout by following the recommendation [here](https://github.com/NVIDIA-NeMo/NeMo/issues/1376#issuecomment-3234432315).

Out of the box it was quite an annoyance to see this every time I loaded a model 😄:
```
[NeMo I 2026-05-06 12:07:53 mixins:184] Tokenizer SentencePieceTokenizer initialized with 1024 tokens
[NeMo W 2026-05-06 12:07:54 modelPT:188] If you intend to do training or fine-tuning, please call the ModelPT.setup_training_data() method and provide a valid configuration file to setup the train data loader.
    Train config : 
    use_lhotse: true
    skip_missing_manifest_entries: true
    input_cfg: null
    tarred_audio_filepaths: null
    manifest_filepath: null
    sample_rate: 16000
    shuffle: true
    num_workers: 2
    pin_memory: true
    max_duration: 40.0
    min_duration: 0.1
    text_field: answer
    batch_duration: null
    use_bucketing: true
    bucket_duration_bins: null
    bucket_batch_size: null
    num_buckets: 30
    bucket_buffer_size: 20000
    shuffle_buffer_size: 10000
    
[NeMo W 2026-05-06 12:07:54 modelPT:195] If you intend to do validation, please call the ModelPT.setup_validation_data() or ModelPT.setup_multiple_validation_data() method and provide a valid configuration file to setup the validation data loader(s). 
    Validation config : 
    use_lhotse: true
    manifest_filepath: null
    sample_rate: 16000
    batch_size: 16
    shuffle: false
    max_duration: 40.0
    min_duration: 0.1
    num_workers: 2
    pin_memory: true
    text_field: answer
    
[NeMo I 2026-05-06 12:07:56 rnnt_models:226] Using RNNT Loss : tdt
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}
[NeMo I 2026-05-06 12:07:56 rnnt_models:226] Using RNNT Loss : tdt
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}
[NeMo W 2026-05-06 12:07:56 label_looping_base:125] No conditional node support for Cuda.
    Cuda graphs with while loops are disabled, decoding speed will be slower
    Reason: CUDA is not available
[NeMo I 2026-05-06 12:07:56 rnnt_models:226] Using RNNT Loss : tdt
    Loss tdt_kwargs: {'fastemit_lambda': 0.0, 'clamp': -1.0, 'durations': [0, 1, 2, 3, 4], 'sigma': 0.02, 'omega': 0.1}
[NeMo W 2026-05-06 12:07:56 label_looping_base:125] No conditional node support for Cuda.
    Cuda graphs with while loops are disabled, decoding speed will be slower
    Reason: CUDA is not available
[NeMo I 2026-05-06 12:07:57 save_restore_connector:285] Model EncDecRNNTBPEModel was successfully restored from /home/ethical_haquer/.cache/huggingface/hub/models--nvidia--parakeet-tdt-0.6b-v2/snapshots/48b630d20b000e5ad3735e5378a2d9bde3f80826/parakeet-tdt-0.6b-v2.nemo.
```

However, I have yet to figure out how to prevent these lines from being logged when I import the library:
```
[NeMo W 2026-05-06 12:05:25 megatron_init:62] Megatron num_microbatches_calculator not found, using Apex version.
OneLogger: Setting error_handling_strategy to DISABLE_QUIETLY_AND_REPORT_METRIC_ERROR for rank (rank=0) with OneLogger disabled. To override: explicitly set error_handling_strategy parameter.
No exporters were provided. This means that no telemetry data will be collected.
```

Any guidance on how to prevent these lines from being logged would be appreciated.

**Comment by anhnguyen-namitech** [comment]
I would like to be able to disable this message:
`OneLogger: Setting error_handling_strategy to DISABLE_QUIETLY_AND_REPORT_METRIC_ERROR for rank (rank=0) with OneLogger disabled. To override: explicitly set error_handling_strategy parameter.
No exporters were provided. This means that no telemetry data will be collected.`

Furthermore, nv_one_logger is a bit brittle. It will insert a hook to on_train_batch_start to start a timer. In our use case, we overwrote on_train_batch_start to return -1 to force a new epoch to reload Lhotse dataloader (https://github.com/Lightning-AI/pytorch-lightning/blob/78bf0214a0ad7571391619dca952c13988c0dc51/src/lightning/pytorch/loops/training_epoch_loop.py#L364). This will crash the training loop because Lightning will skip on_train_batch_end. We have to manually disable nv_one_logger timer via on_training_single_iteration_end() to prevent such issue.
